Manager/Manager.php

```diff
@@ -78,6 +89,32 @@ public function save($entity, array $options = array())
      */
     public function remove($entity)
     {
-        $this->remover->remove($entity);
+        if (!$this->isLinkedToProduct($entity)) {
+            $this->remover->remove($entity);
+        }
```

Comment:
I would move this in the remover. Seems cleaner to me as it is used only there.
What if they are removing from somewhere else (mass delete or custom job, etc.)?!

I think there is some examples in the IVB about that :)

Reply:
:+1: